tests/components/recorder/test_init.py

```diff
@@ -216,8 +215,8 @@ async def test_shutdown_closes_connections(
     instance = recorder.get_instance(hass)
     await instance.async_db_ready
     await hass.async_block_till_done()
-    pool = instance.engine.pool
-    pool.shutdown = Mock()
+    pool = instance.engine
+    pool.dispose = Mock()
```

Comment:
Needs to wrap instead because otherwise it will leave the connection open and the test fails on MySQL